### PR TITLE
Browserstack bitrise espresso step 1.0.0

### PR DESCRIPTION
**Pull request by deepwired**
TagCheck

### What to do if the build fails?

At the moment contributors do not have access to the CI workflow triggered by StepLib PRs. In case of a failed build, we ask for your patience. Maintainers of Bitrise Steplib will sort it out for you or inform you if any further action is needed.

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

**New Step**
Thank you for the new Step share! The CI check might will fail due to our extended validation engine. Nothing to worry about yet, we will get back to you shortly.

### CODE REVIEW

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 1]
```diff
@@ -0,0 +1,256 @@
+title: BrowserStack App Automate - Espresso
+summary: |
+  Run your Espresso tests on BrowserStack App Automate
+description: |
+  Run your Espresso tests on BrowserStack App Automate. This step collects the built APK from $BITRISE_APK_PATH  and test apk from `$BITRISE_TEST_APK_PATH` Environment Variables
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:5]:
```suggestion
  Run your Espresso tests on BrowserStack App Automate. This step collects the built APK from `$BITRISE_APK_PATH`  and test APK from `$BITRISE_TEST_APK_PATH` Environment Variables
```

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    title: BrowserStack username
+- browserstack_accesskey: null
+  opts:
+    category: Authentication
+    description: Access Key of the BrowserStack account
+    is_required: true
+    is_sensitive: true
+    summary: Accesskey for the BrowserStack account
+    title: BrowserStack accesskey
+- android_app_under_test: null
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:51]: Shouldn't the default value of this input be `$BITRISE_APK_PATH`?

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+  opts:
+    category: App & Test Suite
+    description: Path of the app (.apk) file
+    is_expand: true
+    is_required: true
+    summary: Path to the app .apk file
+    title: Android app under test
+- espresso_test_suite: null
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:59]: Same question as above

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    summary: Path to the testsuite .apk file
+    title: Espresso test suite
+- devices_list: null
+  opts:
+    category: Devices
+    description: "Name of one or more device-OS combination in new line \nFor example:\n\n`Samsung
+      Galaxy S9 Plus-9.0`\n\n`Google Pixel 3a-9.0`\n"
+    is_expand: true
+    is_required: true
+    summary: |
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:74]: The summary field is for a one-line sentence summary. I suggest using:

```
Name of one or more device-OS combination in new line from the [device list](https://www.browserstack.com/list-of-browsers-and-platforms/app_automate).
```

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+  opts:
+    category: App & Test Suite
+    description: Path of the test suite (.apk) file
+    is_expand: true
+    is_required: true
+    summary: Path to the testsuite .apk file
+    title: Espresso test suite
+- devices_list: null
+  opts:
+    category: Devices
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:69]: If a category has only a single input, we suggest not using a category at all, it looks cleaner on the UI (unless you have plans for adding more device-related inputs)

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+    is_expand: true
+    is_required: true
+    summary: |
+      Name of one or more device-OS combination in new line from the [device list](https://www.browserstack.com/list-of-browsers-and-platforms/app_automate) . For example:
+
+      `Samsung Galaxy S9 Plus-9.0`
+
+      `Google Pixel 3a-9.0`
+    title: Devices
+- instrumentation_logs: true
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:81]: Boolean input values should be wrapped in quotes (for historical reasons). Please use `"true"` in default values and in the `value_options` too.

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+    category: Test Configuration
+    description: |
+      Provide key-value pairs of filters to run tests: class, package, annotation, size
+
+      For example: class com.foo.AddToCartClass,class com.foo.CheckoutClass
+
+      package com.foo
+
+      Learn more.(https://www.browserstack.com/docs/app-automate/espresso/select-test-cases)
+    summary: "Key-value pairs of filters to run tests from supported test filtering
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:150]: This summary is a bit too long and would look broken on the UI.

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+  opts:
+    category: Test Configuration
+    description: |
+      Provide key-value pairs of filters to run tests: class, package, annotation, size
+
+      For example: class com.foo.AddToCartClass,class com.foo.CheckoutClass
+
+      package com.foo
+
+      Learn more.(https://www.browserstack.com/docs/app-automate/espresso/select-test-cases)
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:149]:
```suggestion
      [Learn more](https://www.browserstack.com/docs/app-automate/espresso/select-test-cases)
```

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+    title: Filter tests
+- opts:
+    category: Test Configuration
+    description: Test sharding splits and run test cases into different group called
+      shards. This behaviour is helpful when your test-suite has a large number of
+      test cases. [Learn more.](https://www.browserstack.com/docs/app-automate/espresso/test-sharding)
+    summary: 'Enable test sharding to split tests cases into different groups instead
+      of running them sequentially. '
+    title: Test sharding
+  use_test_sharding: null
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:165]: Wouldn't it be better as a boolean input with two `value_options`?

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+    summary: Enable to run each test case in a single process to reduce overall build
+      time.
+    title: Single runner invocation
+    value_options:
+    - true
+    - false
+  use_single_runner_invocation: false
+- opts:
+    category: Test Configuration
+    description: A callback URL helps you enable BrowserStack notify you upon the
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:189]: Have you considered marking this input as sensitive?

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+outputs:
+- BROWSERSTACK_BUILD_URL: null
+  opts:
+    description: BrowserStack Dashboard url for the executed build
+    summary: ""
+    title: BrowserStack Dashboard url
+- BROWSERSTACK_BUILD_STATUS: null
+  opts:
+    description: |
+      Status of the executed build. Check values [here:] (https://www.browserstack.com/docs/app-automate/espresso/view-test-results)
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:254]:
```suggestion
      Status of the executed build. Check possible values [here](https://www.browserstack.com/docs/app-automate/espresso/view-test-results)
```

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml [hunk 12]
```diff
@@ -0,0 +1,256 @@
...
+    summary: Bs build status
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:255]:
```suggestion
    summary: BrowserStack build status
```

File: steps/browserstack-bitrise-espresso-step/step-info.yml [hunk 13]
```diff
@@ -0,0 +1 @@
+maintainer: Verified
```
Comment (ofalvai) [inline steps/browserstack-bitrise-espresso-step/step-info.yml:1]:
```suggestion
maintainer: verified
```

This is the correct value I think

**Review by ofalvai — CHANGES_REQUESTED** [review]
Hello @deepwired 👋 

Thank you for contributing to the Bitrise step library 🚀 

I left some comments about step input issues, please take a look.

File: steps/browserstack-bitrise-espresso-step/1.0.0/step.yml (hunk 7 above)
Comment (nitish-bhardwaj-code) [reply steps/browserstack-bitrise-espresso-step/1.0.0/step.yml:150]: "Provide a comma-separated list of class or test names followed by supported [filtering strategies](https://www.browserstack.com/docs/app-automate/espresso/select-test-cases)."

**Pull request closed without merging**